nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/DocumentModel.java

```diff
@@ -848,6 +848,18 @@
      */
     void refresh(int refreshFlags, String[] schemas);
 
+    /**
+     * Moves this document to the destination folder under the given name. If the name is {@code null} or if there is a
+     * collision, a suitable new name is found.
```

Comment:
1. in any case, this should be under a different ticket.
2. up to now all `DocumentModel` methods are acting on just the document itself, never any other. This was a conscious design choice, and I'd like to keep it that way, and force people to use the `CoreSession` if they want to do something cross-documents. If your need is just renaming, I'd be ok with a `rename` method though (name to be discussed).

Reply:
As this was needed mainly for tests and because trash(doc) then untrash(doc) is not an use case we want to address, remove this part and fix tests.